File: Programmers/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.java
import java.util.*;

class Solution {
    public long[] solution(long[] numbers) {
        long[] answer = new long[numbers.length];
        
        for (int i = 0; i < numbers.length; i++) {
            long x = numbers[i];
            
            if ((x & 1) == 0) {
                answer[i] = x + 1;
                continue;
            }
            
            long bit = 1;
            
            while ((x & bit) != 0) {
                bit <<= 1;
            }
            
            answer[i] = x + bit - (bit >> 1);
        }
              
        return answer;
    }
}

File: Programmers/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.cpp
#include <string>
#include <vector>

using namespace std;

vector<long long> solution(vector<long long> numbers) {
    vector<long long> answer;
    answer.reserve(numbers.size());
    
    for (long long x : numbers)
    {
        if ((x & 1LL) == 0)
        {
            answer.push_back(x + 1);
            continue;
        }
        
        long long bit = 1LL;
        
        while (x & bit)
        {
            bit <<= 1;
        }
        
        long long result = x + bit - (bit >> 1);
        answer.push_back(result);
    }
    
    return answer;
}